app/src/main/kotlin/com/fernandocejas/sample/core/interactor/UseCase.kt

```diff
@@ -32,11 +29,15 @@ import kotlinx.coroutines.launch
  */
 abstract class UseCase<out Type, in Params> where Type : Any {
 
+    private val supervisor = SupervisorJob()
+    private val scopeIO = CoroutineScope(Dispatchers.IO + supervisor)
+    private val scopeMain = CoroutineScope(Dispatchers.Main + supervisor)
```

Comment:
Don't you just need `withContext` instead of 2 scopes?

Reply:
Hi @Zhuinden thanks for pointing this out and I think you are right. Do you reckon if I write it this way using `withContext` would this be better? 

> 
>     private val supervisor = SupervisorJob()
>     private val scope = CoroutineScope(Dispatchers.Main + supervisor)
> 
>     abstract suspend fun run(params: Params): Either<Failure, Type>
> 
>     operator fun invoke(params: Params, onResult: (Either<Failure, Type>) -> Unit = {}) {
>         scope.launch {
>             withContext(Dispatchers.IO) {
>                 onResult(run(params))
>             }
>         }
>     }
> 
>     class None

Reply:
Yes